default/scripting/techs/spy/ROOT_DECEPTION.focs.py

```diff
@@ -2,20 +2,116 @@
     BlackHole,
     EffectsGroup,
     InSystem,
+    Min,
     NamedReal,
     Neutron,
+    NoOpCondition,
     NoStar,
     OwnedBy,
     Red,
+    SetSpecialCapacity,
     SetStealth,
     Ship,
     Source,
+    SpecialCapacity,
     Star,
+    Statistic,
     StatisticCount,
+    StatisticIf,
+    Stealth,
     Target,
     Value,
 )
 from focs._tech import *
+from macros.priorities import (
+    AFTER_ALL_TARGET_MAX_METERS_PRIORITY,
+    EARLY_AFTER_ALL_TARGET_MAX_METERS_PRIORITY,
+    LATE_AFTER_ALL_TARGET_MAX_METERS_PRIORITY,
+)
+
+lower_stealth_count_special = "LOWER_STEALTH_COUNT_SPECIAL"
+base_stealth_special = "BASE_STEALTH_SPECIAL"
+
+def count_lower_stealth_ships_statistic_valref():
+    return StatisticCount(
+        float,
+        condition=Ship
+        & InSystem(id=Target.SystemID)
+        & OwnedBy(empire=Source.Owner)
+        & (Value(Target.Stealth) >= Value(LocalCandidate.Stealth)),
+    )
+
+target_has_less_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    & (Value(Target.Stealth) < Value(LocalCandidate.Stealth))
+)
+
+target_has_less_or_equal_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    & (SpecialCapacity(name=base_stealth_special,object=Target.ID) <= SpecialCapacity(name=base_stealth_special,object=LocalCandidate.ID)))
+
+target_has_equal_or_more_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    #& (Value(Target.Stealth) >= Value(LocalCandidate.Stealth))
+    & (SpecialCapacity(name=base_stealth_special,object=Target.ID) >= SpecialCapacity(name=base_stealth_special,object=LocalCandidate.ID))
+)
+
+# works: there are ships which have less stealth
+target_has_more_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    #& (Value(Target.Stealth) > Value(LocalCandidate.Stealth))
+    & (SpecialCapacity(name=base_stealth_special,object=Target.ID) > SpecialCapacity(name=base_stealth_special,object=LocalCandidate.ID))
+)
+
+# works: there are ships which have more stealth than the candidate
+candidate_has_less_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    & (SpecialCapacity(name=base_stealth_special,object=Target.ID) < SpecialCapacity(name=base_stealth_special,object=LocalCandidate.ID))
+    #& ~( Ship & InSystem(id=Target.SystemID) & OwnedBy(empire=Source.Owner)
+    #     & (SpecialCapacity(name=base_stealth_special,object=Target.ID) < SpecialCapacity(name=base_stealth_special,object=LocalCandidate.ID)))
+)
+
+# there are no ships which have more stealth
+target_has_most_stealth_cond = (
+    Ship
+    & InSystem(id=Target.SystemID)
+    & ~IsTarget
+    & OwnedBy(empire=Source.Owner)
+    & ~candidate_has_less_stealth_cond
+)
+
+bla = "bla"
```

Comment:
```suggestion
bla = "bla"  # bla
```

Not sure why you need it